File: frontend/lib/models/menu_item.dart
class MenuItem {
  final String id;
  final String name;
  final String description;
  final double price;
  final String category;
  final String sku;
  final bool available;
  final List<String> tags;
  final int? stock;

  MenuItem({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.category,
    required this.sku,
    required this.available,
    required this.tags,
    required this.stock,
  });

  factory MenuItem.fromJson(Map<String, dynamic> j) => MenuItem(
        id: j['_id'] as String,
        name: j['name'] as String,
        description: (j['description'] ?? '') as String,
        price: (j['price'] as num).toDouble(),
        category: j['category'] as String,
        sku: (j['sku'] ?? '') as String,
        available: (j['available'] ?? true) as bool,
        tags: (j['tags'] as List?)?.map((e) => e.toString()).toList() ?? [],
        stock: j['stock'] == null ? null : (j['stock'] as num).toInt(),
      );
}

File: backend/src/models/MenuItem.js
const mongoose = require('mongoose');

const menuItemSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true, index: true },
    description: { type: String, default: '' },
    price: { type: Number, required: true, min: 0 },
    category: { type: String, required: true, trim: true, index: true },
    sku: { type: String, trim: true, index: true },
    available: { type: Boolean, default: true },
    tags: [{ type: String, trim: true }],
    imageUrl: { type: String, default: '' },
    // Lightweight inventory hooks — real inventory would have a separate collection.
    stock: { type: Number, default: null }, // null = not tracked
    outletId: { type: String, default: 'default', index: true },
  },
  { timestamps: true }
);

menuItemSchema.index({ name: 'text', description: 'text', tags: 'text' });

module.exports = mongoose.model('MenuItem', menuItemSchema);
